app/models/media_resource.rb

```diff
@@ -4,10 +4,36 @@ class MediaResource < ApplicationRecord
 
   belongs_to :episode, -> { with_deleted }, touch: true, optional: true
 
+  acts_as_paranoid
+
   enum status: [:started, :created, :processing, :complete, :error, :retrying, :cancelled]
 
   before_validation :initialize_attributes, on: :create
 
+  after_create :replace_resources!
+
+  scope :complete_or_replaced, -> do
+    with_deleted
+      .complete
+      .where("deleted_at IS NULL OR replaced_at IS NOT NULL")
+      .order("created_at DESC")
+  end
```

Comment:
Does this need to be wrapped in e.g. read committed semantics when saving the form and reading via this scope?  The nested form model save _should_ be in a read-committed transaction, but I'm not sure about the reading side here.

Reply:
Yep you are right!  The episode save wraps the nested models' saves [in a transaction with read-committed isolation](https://api.rubyonrails.org/v7.0.4.2/classes/ActiveRecord/Callbacks.html#module-ActiveRecord::Callbacks-label-Transactions). Otherwise readers would see the positions shifting around as each Content's after_create hooks run.